clients/fides-js/src/lib/cookie.ts

```diff
@@ -289,18 +222,17 @@ export const updateExperienceFromCookieConsentNotices = ({
   cookie: FidesCookie;
   debug?: boolean;
 }): PrivacyExperience => {
-  const noticesWithConsent = experience.privacy_notices?.map((notice) => {
-    // Prefers preference in cookie if it exists, else uses current preference on the notice if it exists, else uses
-    // undefined. Undefined will occur for server-side-fetched experience when no corresponding pref exists in cookie.
-    const defaultPreference = notice.current_preference ?? undefined;
-    const preference = Object.hasOwn(cookie.consent, notice.notice_key)
-      ? transformConsentToFidesUserPreference(
-          Boolean(cookie.consent[notice.notice_key]),
-          notice.consent_mechanism
-        )
-      : defaultPreference;
-    return { ...notice, current_preference: preference };
-  });
+  // todo- instead of updating experience here, push this logic into UI
```

Comment:
are you still planning on doing this? I think it'd be nice to remove `PrivacyNoticeWithPreference` entirely if possible

Reply:
Given the timing of the release, I think we defer this if possible to a later refactor. I'll make a ticket for it, with the tech debt discussion in mind from last retro!

Reply:
this is also related to our ability to get rid of the `PrivacyNoticeWithPreference` type, as this is the only place where it's needed- when we assign it on the experience here. I'll make note of these things in the ticket

Reply:
https://ethyca.atlassian.net/browse/PROD-1568